Network/NetworkService.swift

```diff
@@ -63,4 +72,34 @@ class NetworkService {
     func cancel() {
         task?.cancel()
     }
+    
+    
+    //MARK: Private
+    private func makeQuery(for url: URL, params: [String: Any], type: QueryType) -> URLRequest? {
```

Comment:
Wondering, why method is returning optional value instead of crash in case something went wrong? IMO it should be consider as programmer bug when something goes wrong in this specific method. What you think?